service/controller/key/key.go

```diff
@@ -1,7 +1,25 @@
 package key
 
+import (
+	"github.com/giantswarm/microerror"
+	apiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
 // DNSIP returns the IP of the DNS service given a cluster IP range.
 func DNSIP(clusterIPRange string) (string, error) {
 	// TODO Add logic.
 	return "", nil
 }
+
+func ToCluster(v interface{}) (apiv1alpha3.Cluster, error) {
```

Comment:
Can we return a pointer type instead? 
```suggestion
func ToCluster(v interface{}) (*apiv1alpha3.Cluster, error) {
```

Reply:
We return the type in ToApp and ToChart so I'd prefer to keep this consistent with that.

https://github.com/giantswarm/app/blob/5835ab9edf8f4e8c93fce902b569068702a8668e/pkg/key/app.go#L152